src/box/memtx_allocator.h

```diff
@@ -331,6 +331,32 @@ class MemtxAllocator {
 		}
 	}
 
+	/**
+	 * Fill `tuple_info' with the allocator-specific information about the
+	 * `tuple'.
+	 */
+	static void tuple_info(struct tuple *tuple,
+			       struct tuple_info *tuple_info)
+	{
+		size_t size = tuple_size(tuple);
+		struct small_alloc_info alloc_info;
+		Allocator::get_alloc_info(tuple, size, &alloc_info);
+
+		if (alloc_info.is_large) {
+			tuple_info->waste_size = 0;
+			tuple_info->arena_type = TUPLE_ARENA_MALLOC;
+		} else {
+			tuple_info->waste_size = alloc_info.real_size - size;
+			const char *alloc_name = Allocator::get_name();
+			if (strcmp(alloc_name, "SmallAlloc") == 0)
+				tuple_info->arena_type = TUPLE_ARENA_MEMTX;
+			else if (strcmp(alloc_name, "SysAlloc") == 0)
+				tuple_info->arena_type = TUPLE_ARENA_MALLOC;
```

Comment:
Comparing strings when we have numeric constants looks ugly. Can't `get_alloc_info` return numeric constants instead of strings? Even better, why not set the allocator type at the upper level, in `memtx_tuple_info` (like we do in `runtime_tuple_info` and `vy_tuple_info`) and drop `Allocator::get_name` altogether? Then we wouldn't need `MemtxAllocator::tuple_info` either since we could use `ALLOC::tuple_info` directly in `memtx_tuple_info`.

Reply:
Done.